subprojects/packagefiles/theora/lib/meson.build

```diff
@@ -0,0 +1,204 @@
+
+                'x86/sse2fdct.c',
+
+                'x86/mmxfrag.c',
+                'x86/mmxidct.c',
+                'x86/mmxstate.c',
+                'x86/x86state.c',
+            )
+            decoder_sources += files(
+                'x86/mmxidct.c',
+                'x86/mmxfrag.c',
+                'x86/mmxstate.c',
+                'x86/x86state.c',
+            )
+        endif
+    elif conf.get('CPU_x86_32', false)
+        if cc.get_id() != 'msvc'
+            decoder_sources += files(
+                'x86/x86cpu.c',
+                'x86/mmxidct.c',
+                'x86/mmxfrag.c',
+                'x86/mmxstate.c',
+                'x86/sse2idct.c',
+                'x86/x86state.c',
+            )
+            encoder_sources += files(
+                'x86/mmxencfrag.c',
+                'x86/mmxfdct.c',
+                'x86/x86enc.c',
+
+                'x86/x86cpu.c',
+                'x86/mmxfrag.c',
+                'x86/mmxidct.c',
+                'x86/mmxstate.c',
+                'x86/x86state.c',
+            )
+        else
+            encoder_sources += files(
+                'x86_vc/mmxencfrag.c',
+                'x86_vc/mmxfdct.c',
+                'x86_vc/x86enc.c',
+                'x86_vc/mmxfrag.c',
+                'x86_vc/mmxidct.c',
+                'x86_vc/mmxstate.c',
+                'x86_vc/x86state.c',
+            )
+            decoder_sources += files(
+                'x86_vc/mmxidct.c',
+                'x86_vc/mmxfrag.c',
+                'x86_vc/mmxstate.c',
+                'x86_vc/x86state.c',
+            )
+        endif
+    endif
+endif
+
+decoder_res = files()
+encoder_res = files()
+
+if host_machine.system() == 'windows'
+    if get_option('buildtype') == 'debug'
+        decoder_rc = '../win32/xmingw32/libtheoradecd.rc'
+        encoder_rc = '../win32/xmingw32/libtheoraencd.rc'
+    else
+        decoder_rc = '../win32/xmingw32/libtheoradec.rc'
+        encoder_rc = '../win32/xmingw32/libtheoraenc.rc'
+    endif
+
+    patcher_py = find_program(
+        '../win32/xmingw32/generate_windows_rc.py',
+        required: true, native: true
+    )
+
+    libtheoradec_rc = custom_target(
+        'libtheoradec.rc',
+        command: [patcher_py, '@INPUT@', '@OUTPUT@', meson.project_version()],
+        input: [decoder_rc, '../win32/xmingw32/libtheoradec-all.rc'],
+        output: ['libtheoradec.rc']
+    )
+
+    decoder_res += windows.compile_resources(libtheoradec_rc)
+
+    libtheoraenc_rc = custom_target(
+        'libtheoraenc.rc',
+        command: [patcher_py, '@INPUT@', '@OUTPUT@', meson.project_version()],
+        input: [encoder_rc, '../win32/xmingw32/libtheoraenc-all.rc'],
+        output: ['libtheoraenc.rc']
+    )
+
+    encoder_res += windows.compile_resources(libtheoraenc_rc)
+endif
+
+if host_machine.system() == 'darwin'
+    version_flag = '-Wl,-exported_symbols_list,@0@'
+    theoradec_version_arg = join_paths(meson.current_source_dir(), 'theoradec.exp')
```

Comment:
an alternative to join_paths is to use the / operator.

Reply:
Much cleaner, thanks!